src/note/asset_base.rs

```diff
@@ -101,7 +102,7 @@ impl AssetBase {
     /// Generates a ZSA random asset.
     ///
     /// This is only used in tests.
-    pub(crate) fn random(rng: &mut impl RngCore) -> Self {
+    pub(crate) fn random(rng: &mut (impl RngCore + CryptoRngCore)) -> Self {
```

Comment:
`RngCore` is not necessary
`pub(crate) fn random(rng: &mut impl CryptoRngCore)

Reply:
Fixed.